lib/trento/application/usecases/hosts/heartbeats.ex

```diff
@@ -37,17 +39,19 @@ defmodule Trento.Heartbeats do
     |> Repo.transaction()
   end
 
-  @spec dispatch_heartbeat_failed_commands :: :ok
-  def dispatch_heartbeat_failed_commands do
-    Enum.each(get_all_expired_heartbeats(), fn %{agent_id: agent_id} ->
+  @spec dispatch_heartbeat_failed_commands(DateService) :: :ok
+  def dispatch_heartbeat_failed_commands(data_service \\ DateService) do
+    Enum.each(get_all_expired_heartbeats(data_service), fn %{agent_id: agent_id} ->
       dispatch_command(agent_id, :critical)
     end)
   end
 
-  defp get_all_expired_heartbeats do
+  defp get_all_expired_heartbeats(data_service) do
```

Comment:
`s/data_service/date_service/g` :smile:

Reply:
Also, is it fine to have `DateService` in the spec? Shouldn't it be `DateService.t()`?

Reply:
yes, thanks. updating

Reply:
Thanks for the explanation!

Reply:
Amended